pkg/retrier/retrier.go

```diff
@@ -81,8 +86,17 @@ func (r *Retrier) Retry(fn func() error) error {
 			return err
 		}
 		if r.backoffFactor != nil {
-			wait = wait * time.Duration(*r.backoffFactor*float32(retries))
+			wait = time.Duration(float32(wait) * (*r.backoffFactor * float32(retries)))
```

Comment:
Truncation of the fractional portion in the casting to int types was making the calculated wait times inaccurate

Reply:
Thanks for the detail